jetstream/kv.go

```diff
@@ -174,10 +174,17 @@ type (
 		// Deprecated: Use ListKeys instead to avoid memory issues.
 		Keys(ctx context.Context, opts ...WatchOpt) ([]string, error)
 
+		// KeysWithFilters returns a filtered list of keys in the bucket.
+		// It returns a complete slice of matching keys.
+		KeysWithFilters(ctx context.Context, kv KeyValue, filters ...string) ([]string, error)
```

Comment:
As per my comment from earlier, we decided to stick with having a single method (at least for now) returning a `KeyLister`, so this one should be removed.